### PR TITLE
Feature: Allowed users to set custom ringtone

### PR DESCRIPTION
**Pull request by tavkeer**
### Description
previously users were not able to set custom ringtone for the alarm now using file picker they would select the desired audio file and set it as ringtone

### Proposed Changes
achieved this using file picker to pick file and then used audioPlayers to play the audio also used hive to store the file location

## Checklist

- [x] Tests have been added or updated to cover the changes
- [ ] Documentation has been updated to reflect the changes
- [x] Code follows the established coding style guidelines
- [x] All tests are passing

### CODE REVIEW

**Comment by tavkeer** [comment]
@MarkisDev 
here is a video showing the change!

**Comment by tavkeer** [comment]
@MarkisDev 

https://github.com/CCExtractor/ultimate_alarm_clock/assets/105163810/689e1a1e-a83f-4993-b34e-7f0dc942bfed

**Comment by tavkeer** [comment]
@MarkisDev 
this runs smoothly!

File: pubspec.yaml [hunk 1]
```diff
@@ -6,47 +6,50 @@ environment:
...
+  flutter_map: ^4.0.0
   flutter_ringtone_player: ^3.2.0
+  flutter_secure_storage: ^8.0.0
+  flutter_svg: ^2.0.4
   flutter_time_picker_spinner: ^2.0.0
-  cloud_firestore: ^4.4.5
+  get: 4.6.5
+  google_sign_in: ^6.0.2
+  hive: ^2.2.3
+  hive_flutter: ^1.1.0
```
Comment (MarkisDev) [inline pubspec.yaml:31]: Why another database? We already have IsarDB and Secure Storage which can be used to store the path...

**Review by MarkisDev — CHANGES_REQUESTED** [review]

**Comment by MarkisDev** [comment]
This is great! I love that we can have custom ringtones now - however, we need to keep it as an option for every alarm being set and have a default ringtone setting as well.

Eventually, we need to be able to give some default choices as well, so that the user doesn't have to always download most commonly used alarm ringtones.

File: pubspec.yaml (hunk 1 above)
Comment (tavkeer) [reply pubspec.yaml:31]: I could have missed it will implement that in next after implying the suggested changes

**Comment by tavkeer** [comment]
> This is great! I love that we can have custom ringtones now - however, we need to keep it as an option for every alarm being set and have a default ringtone setting as well.
> 
> Eventually, we need to be able to give some default choices as well, so that the user doesn't have to always download most commonly used alarm ringtones.

Hey there I am aiming for GSOC 2024 can you mentor me
That would be a great help

**Comment by MarkisDev** [comment]
@tavkeer I'm happy to help you to the best of my knowledge! You should definitely reach me on Slack - there are so many more experienced developers in the CCExtractor community who will help you!

**Comment by MarkisDev** [comment]
Also, what's the progress on this PR? Are you facing any difficulty with making the ringtone individual?

**Pull request closed without merging**